src/Avalonia.DesignerSupport/DesignerEventConverter.cs

```diff
@@ -0,0 +1,87 @@
+            {
+                var rootObjectProvider = context.GetService(typeof(IRootObjectProvider)) as IRootObjectProvider;
+                var destinationTypeProvider = context.GetService(typeof(IDestinationTypeProvider)) as IDestinationTypeProvider;
+                if (rootObjectProvider != null && destinationTypeProvider != null)
+                {
+                    var target = rootObjectProvider.RootObject;
+                    var eventType = destinationTypeProvider.GetDestinationType();
+                    var eventParameters = eventType.GetRuntimeMethods().First(r => r.Name == "Invoke").GetParameters();
+                    // go in reverse to match System.Xaml behaviour
+                    var methods = target.GetType().GetRuntimeMethods().Reverse();
+
+                    // find based on exact match parameter types first
+                    foreach (var method in methods)
+                    {
+                        if (method.Name != text)
+                            continue;
+                        var parameters = method.GetParameters();
+                        if (eventParameters.Length != parameters.Length)
+                            continue;
+                        if (parameters.Length == 0)
+                            return method.CreateDelegate(eventType, target);
+
+                        for (int i = 0; i < parameters.Length; i++)
+                        {
+                            var param = parameters[i];
+                            var eventParam = eventParameters[i];
+                            if (param.ParameterType != eventParam.ParameterType)
+                                break;
+                            if (i == parameters.Length - 1)
+                                return method.CreateDelegate(eventType, target);
+                        }
+                    }
+
+                    // EnhancedXaml: Find method with compatible base class parameters
+                    foreach (var method in methods)
+                    {
+                        if (method.Name != text)
+                            continue;
+                        var parameters = method.GetParameters();
+                        if (parameters.Length == 0 || eventParameters.Length != parameters.Length)
+                            continue;
+
+                        for (int i = 0; i < parameters.Length; i++)
+                        {
+                            var param = parameters[i];
+                            var eventParam = eventParameters[i];
+                            if (!param.ParameterType.GetTypeInfo().IsAssignableFrom(eventParam.ParameterType.GetTypeInfo()))
+                                break;
+                            if (i == parameters.Length - 1)
+                                return method.CreateDelegate(eventType, target);
+                        }
+                    }
+
+                    // We want to ignore missing events in the designer, so if event handler
```

Comment:
It's probably better to check for https://github.com/AvaloniaUI/Avalonia/blob/master/src/Markup/Avalonia.Markup.Xaml/PortableXaml/AvaloniaXamlSchemaContext.cs#L18 rather than rely on special converter being registered in DesignWindowLoader static constructor. Static constructor might be called in non-design mode and we'll end up ignoring missing event handlers.

Reply:
Good point about the static constructor. I'll move the registration out of that.

The reason I used a special converter however is that I didn't want to hard-code design-time behavior into `Avalonia.Markup.Xaml`. Don't you think this is a good idea? Would you prefer explicit design mode support to be added to `Avalonia.Markup.Xaml`?

Reply:
>Would you prefer explicit design mode support

It's already there - https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.DesignerSupport/DesignWindowLoader.cs#L21

Reply:
What I mean is: you say:

> rather than rely on special converter being registered in DesignWindowLoader static constructor.

Are you suggesting that instead of checking for a type converter in `AvaloniaMemberAttributeProvider`, that I should hard-code using the `DesignerEventConverter` in `Avalonia.Markup.Xaml` when `IsDesignMode == true`?

The reason I didn't do it like this is that we don't know how other designers will want to handle this case. By making it an extension point we're allowing the designer implementation to choose how to handle event handlers.

Reply:
I'm not sure what do you mean by "other designers". The upcoming compiler doesn't use converters for event subscriptions at all, it just emits `target.Event += context.RootObject.Method;`

Reply:
By "other designers" I just mean if someone comes along and has a great idea for a designer and wants to handle events differently. It's just feeling that we should expose extension points in something as general as `Avalonia.Markup.Xaml` rather than hard-coding designer behavior.

However I'm not tied to doing it like this and I'm unclear as to what you think I should do instead exactly. Please tell me how you think this should be done.

Reply:
I'd say, let's solve the problems when they arise. Portable.Xaml will be removed this year and compiler has completely different mechanism for event subscriptions.

Reply:
Ok, updated to do it like you suggested. I also added unit tests, which required using a non-shared `AvaloniaXamlSchemaContext` as otherwise the `IsDesignTime` setting interfered with other tests.

Reply:
>non-shared AvaloniaXamlSchemaContext

This could be a huge performance hit. Might be worth to keep two contexts (design and non-design one) for now.

Reply:
You're right of course. I saw that an instance of `IRuntimeTypeProvider` was being passed to `AvaloniaXamlSchemaContext` and assumed that that would do the caching of types/members etc. Looking more it seems not. Like you say keeping two contexts might be the best hack for now until the XAML compiler.

Reply:
Just pushed a fix for this.